### PR TITLE
FRW-8618 Initial commit

### PR DESCRIPTION
**Pull request by dimitriyTsemma**
## PR Description

TBD

## Checklist
- [x] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: src/Spryker/Service/OtelElasticaInstrumentation/OpenTelemetry/ElasticaInstrumentation.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+    /**
+     * @var string
+     */
+    protected const SPAN_NAME = 'elasticsearch';
+
+    /**
+     * @var string
+     */
+    protected const HOST = 'host';
```
Comment (gechetspr) [inline src/Spryker/Service/OtelElasticaInstrumentation/OpenTelemetry/ElasticaInstrumentation.php:36]: could you name those constant a bit more descriptive? E.g. HEADER_HOST in this case

File: src/Spryker/Service/OtelElasticaInstrumentation/OpenTelemetry/ElasticaInstrumentation.php [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                    ->setAttribute(TraceAttributes::URL_FULL, $request->getRequest()->getUri())
+                    ->setAttribute(TraceAttributes::HTTP_REQUEST_METHOD, $request->getRequest()->getMethod())
+                    ->setAttribute(TraceAttributes::URL_QUERY, $request->getRequest()->getQueryString())
+                    ->setAttribute(TraceAttributes::URL_DOMAIN, $request->getRequest()->headers->get(static::HOST))
+                    ->startSpan();
+                $span->activate();
+
+                Context::storage()->attach($span->storeInContext(Context::getCurrent()));
+            },
+            post: function ($instance, array $params, $response, ?Throwable $exception): void {
```
Comment (gechetspr) [inline src/Spryker/Service/OtelElasticaInstrumentation/OpenTelemetry/ElasticaInstrumentation.php:69]: I assume we can use Client typehint here too

File: src/Spryker/Service/OtelElasticaInstrumentation/OpenTelemetry/ElasticaInstrumentation.php [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                $instrumentation = new CachedInstrumentation();
+                $request = new RequestProcessor();
+
+                $span = $instrumentation::getCachedInstrumentation()
+                    ->tracer()
+                    ->spanBuilder(static::SPAN_NAME)
+                    ->setSpanKind(SpanKind::KIND_CLIENT)
+                    ->setAttribute(TraceAttributes::URL_FULL, $request->getRequest()->getUri())
+                    ->setAttribute(TraceAttributes::HTTP_REQUEST_METHOD, $request->getRequest()->getMethod())
+                    ->setAttribute(TraceAttributes::URL_QUERY, $request->getRequest()->getQueryString())
```
Comment (gechetspr) [inline src/Spryker/Service/OtelElasticaInstrumentation/OpenTelemetry/ElasticaInstrumentation.php:62]: Is it enough to catch the elasticsearch query? I mean isn't it a query for the root request?

File: src/Spryker/Service/OtelElasticaInstrumentation/OtelElasticaInstrumentationService.php [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+ */
+
+namespace Spryker\Service\OtelElasticaInstrumentation;
+
+use Spryker\Service\Kernel\AbstractService;
+
+/**
+ * @method \Spryker\Service\OtelElasticaInstrumentation\OtelElasticaInstrumentationServiceFactory getFactory()
+ */
+class OtelElasticaInstrumentationService extends AbstractService implements OtelElasticaInstrumentationServiceInterface
```
Comment (gechetspr) [inline src/Spryker/Service/OtelElasticaInstrumentation/OtelElasticaInstrumentationService.php:15]: we don't need it. This is not a module and will not use Spryker architecture. It should be as lean as possible.

I guess Kernel dependency came from it

**Review by gechetspr — CHANGES_REQUESTED** [review]

File: src/Spryker/Service/OtelElasticaInstrumentation/OpenTelemetry/ElasticaInstrumentation.php [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+     * @return void
+     */
+    public static function register(): void
+    {
+        // phpcs:disable
+        hook(
+            class: Client::class,
+            function: static::METHOD_NAME,
+            pre: function (Client $client, array $params): void {
+                $instrumentation = new CachedInstrumentation();
```
[the remark this answers is not in the document]
Comment (dimitriyTsemma) [reply src/Spryker/Service/OtelElasticaInstrumentation/OpenTelemetry/ElasticaInstrumentation.php:53]: I grabbed it from another OTEL packages 
but this class does the following
```
/**
     * @var string
     */
    protected const INSTRUMENTATION_NAME = 'io.opentelemetry.contrib.php.spryker';

    /**
     * @var string
     */
    protected const INSTRUMENTATION_VERSION = '1.0.0';

    /**
     * @var \OpenTelemetry\API\Instrumentation\CachedInstrumentation|null
     */
    protected static ?OpenTelemetryCachedInstrumentation $instrumentation = null;

    /**
     * @return \OpenTelemetry\API\Instrumentation\CachedInstrumentation|null
     */
    public static function getCachedInstrumentation(): ?OpenTelemetryCachedInstrumentation
    {
        if (static::$instrumentation === null) {
            static::setCachedInstrumentation();
        }

        return static::$instrumentation;
    }

    /**
     * @return void
     */
    protected static function setCachedInstrumentation(): void
    {
        static::$instrumentation = (new OpenTelemetryCachedInstrumentation(
            static::INSTRUMENTATION_NAME,
            static::INSTRUMENTATION_VERSION,
        ));
    }
```

[the remark this answers is not in the document]
Comment (gechetspr) [reply src/Spryker/Service/OtelElasticaInstrumentation/OpenTelemetry/ElasticaInstrumentation.php:53]: I get it :) I mean there are only static methods. We don't need a class instance itself

File: src/Spryker/Service/OtelElasticaInstrumentation/OpenTelemetry/ElasticaInstrumentation.php (hunk 3 above)
Comment (dimitriyTsemma) [reply src/Spryker/Service/OtelElasticaInstrumentation/OpenTelemetry/ElasticaInstrumentation.php:62]: image

Comment (gechetspr) [reply src/Spryker/Service/OtelElasticaInstrumentation/OpenTelemetry/ElasticaInstrumentation.php:61]: Could you also check the storefront?

**Review by gechetspr — APPROVED** [review]

**Review by gechetspr — APPROVED** [review]

**Pull request merged**